codes/csharp/chapter_graph/graph_adjacency_list.cs

```diff
@@ -29,28 +29,31 @@ public class GraphAdjList {
 
     /* 添加边 */
     public void AddEdge(Vertex vet1, Vertex vet2) {
-        if (!adjList.ContainsKey(vet1) || !adjList.ContainsKey(vet2) || vet1 == vet2)
+        if (!adjList.TryGetValue(vet1, out List<Vertex>? vectors1)
+            || !adjList.TryGetValue(vet2, out List<Vertex>? vectors2)
+            || vet1 == vet2)
             throw new InvalidOperationException();
-        // 添加边 vet1 - vet2
-        adjList[vet1].Add(vet2);
-        adjList[vet2].Add(vet1);
+        vectors1.Add(vet2);
+        vectors2.Add(vet1);
     }
 
     /* 删除边 */
     public void RemoveEdge(Vertex vet1, Vertex vet2) {
-        if (!adjList.ContainsKey(vet1) || !adjList.ContainsKey(vet2) || vet1 == vet2)
+        if (!adjList.TryGetValue(vet1, out List<Vertex>? vectors1)
```

Comment:
A cool new feature, but less readable for me.

Reply:
It's not a new feature, just to avoid double lookup for a dictionary key, revert change.